### ISSUE TITLE
linter: `eslint(no-unused-vars)` false positive when namespace is shadowed by generic type parameter

### ISSUE THREAD

**Issue by rubenferreira97** [oxc-project/oxc#17927]
I encountered a false positive in `eslint(no-unused-vars)` where a namespace is flagged as unused if it is shadowed by a generic type parameter in a function signature, even though the namespace is explicitly used in the type constraint.

This is a confirmed bug in upstream `typescript-eslint`: **[Issue #10746](https://github.com/typescript-eslint/typescript-eslint/issues/10746)**.

Oxc currently faithfully reproduces this buggy behavior.

### Reproduction

**[Oxc Playground Link](https://playground.oxc.rs/#eNptVE1r3DAQ/StGpxbckgYKwUkDJelCIW3SJjSXvcj2eKNEllxpnM1i/N/7ZFvaLezJ0puvN29GHkQlCmFkS76TFWXXkmUpPWXD2mQZvXXWcca7jrIHWWq6eLjMvmQP52szrs3aLPamNxUrazImzxcxRT6HIAmTqX1K/XHOFK+Xl+9CqXq5FskvDzCOGtgUsjbvi2w44nn+n1s2zuQdce/MQUA+u2VjoC9yYUUxCNeb8NHKsCjY9ZSLxrpWMpMTRSM14gQ7aXyAE6K81ZKpvqZKSydD9z4Zt08KSgQ9E9RKs9H7a2XbzpHfh/jKdjDPBPyuLa2Ot6rZLG5jLjrpfCA2iElWj7qYH/s39CO1ttvfU9e3PXtV02oZTEwVapJ7pTvpEBvRKe7P2XfDTiFjlQyeQJtV9c056xYUHIJWEwecD7QaRO8JE9g3BbkfVc1PojjNBXbgtrlRBl0K3YBuh2q82M9OUE0FiX71lvc6Pfu3+yPw33C7c7ZDLSH9B0NUU42cGJQCvc2VbVsZbFoDRR8qNYtetrF/2Ldy5+FSOlm9EN9jaAiPzhHF85iZL/Vt+UwVPzrZIUcUNRSauH5lCFn2YEjuIAxqpT2a9WLpNoS1E+RPTz59RgIoeE0NYlbWXWnp/UqRrtNA8N7IqZYMS43Ns1i8NBhYW8UJ/UEsw+KnqR3f2EF4kO0we3JGYuci1crWtKHpbZgw47SOz762+GcsahpjecoVEU2b6DtliXseliU+AnRuuxt6pZT0haj7GX5ChwSwj1avsJsh4pVcaf2BlsqEGQRTOo2A60m+AKfTGPAgc+xp/AcSy8O/)**

```typescript
namespace Database {
  export type Table<T> = T;
}

// 'Database' generic shadows the namespace, but namespace is used in 'Database.Table'
export function test<Database, Table extends Database.Table<Database>>(
  database: Database,
  table: Table
): { database: Database; table: Table } {
  return { database, table };
}

```

### Actual Behavior

The linter reports `Database` (the namespace) as unused:

```text
  ! eslint(no-unused-vars): Variable 'Database' is declared but never used.
   ,-[:1:11]
 1 | namespace Database {
   :           ^^^^|^^^
   :               `-- 'Database' is declared here
 2 |   export type Table<T> = T;
   `----
  help: Consider removing this declaration.

```

### Expected Behavior

The namespace `Database` should be marked as used because it is accessed in the generic constraint `extends Database.Table<...>`.

### Question regarding Scope

Since this is an upstream bug that Oxc has successfully replicated: **Is the goal here to fix this logic in Oxc to be correct, or is 1:1 compatibility with ESLint (including reproducing their bugs) the priority?**

**Comment by camc314** [comment]
fixed in #17930 / #17929
